service/rtc/track.go

```diff
@@ -15,6 +15,8 @@ const (
 )
 
 type trackActionContext struct {
-	action trackAction
-	track  webrtc.TrackLocal
+	action        trackAction
+	localTrack    webrtc.TrackLocal
+	remoteTrack   *webrtc.TrackRemote
+	senderSession *session
```

Comment:
Could I get an explanation for the change here (which is throughout, really) -- since the localTrack and remoteTrack differentiation is new, I'm just wondering why we didn't need it before and why we need it now? (I think my context window for the sfu blew past my token limit a little while ago)

Reply:
Good question. These are useful so we can avoid a bunch of complexity in `handleSenderRTCP`. For screen sharing, it was simple enough to track things on the `call` object because we had the guarantee of one screen track/session globally per call at any given time, but now we could be handling concurrent incoming video tracks so it's a bit simpler to do it this way since we only need this info in there.

https://github.com/mattermost/rtcd/blob/7bc1f7ac6460f11520335e60f075a0acd9aa3f02/service/rtc/session.go#L330-L351

Reply:
Ah, I see. Yep, good idea to keep it separate, I like it.